src/core/functions/aggregate/llm_min_or_max.cpp

```diff
@@ -0,0 +1,210 @@
+#include <flockmtl/core/functions/aggregate/llm_min_or_max.hpp>
+#include <flockmtl/core/functions/prompt_builder.hpp>
+#include <templates/llm_min_or_max_prompt_template.hpp>
+#include "flockmtl/core/module.hpp"
+#include "flockmtl/core/model_manager/model_manager.hpp"
+#include "flockmtl/core/model_manager/tiktoken.hpp"
+
+namespace flockmtl {
+namespace core {
+
+void LlmMinOrMaxState::Initialize() {
+}
+
+void LlmMinOrMaxState::Update(const nlohmann::json &input) {
+    value.push_back(input);
+}
+
+void LlmMinOrMaxState::Combine(const LlmMinOrMaxState &source) {
+    value = std::move(source.value);
+}
+
+LlmMinOrMax::LlmMinOrMax(std::string &model, int model_context_size, std::string &user_prompt,
+                         std::string &llm_min_or_max_template)
+    : model(model), model_context_size(model_context_size), user_prompt(user_prompt),
+      llm_min_or_max_template(llm_min_or_max_template) {
+
+    int fixed_tokens = calculateFixedTokens();
+
+    if (fixed_tokens > model_context_size) {
+        throw std::runtime_error("Fixed tokens exceed model context size");
+    }
+
+    available_tokens = model_context_size - fixed_tokens;
+}
+
+int LlmMinOrMax::calculateFixedTokens() const {
+    int fixed_tokens = 0;
+    fixed_tokens += Tiktoken::GetNumTokens(user_prompt);
+    fixed_tokens += Tiktoken::GetNumTokens(llm_min_or_max_template);
+    return fixed_tokens;
+}
+
+nlohmann::json LlmMinOrMax::GetElement(const nlohmann::json &tuples) {
+    inja::Environment env;
+    nlohmann::json data;
+    data["tuples"] = tuples;
+    data["user_prompt"] = user_prompt;
+    auto prompt = env.render(llm_min_or_max_template, data);
+
+    nlohmann::json settings;
+    auto response = ModelManager::CallComplete(prompt, model, settings);
+    return response["selected"];
+}
+
+nlohmann::json LlmMinOrMax::LlmMinOrMaxCall(nlohmann::json &tuples) {
+
+    if (tuples.size() == 1) {
```

Comment:
Not a big fan of early returns but we can keep it as is for now.

Reply:
done